### PR TITLE
Use `copytree` in `pack`

### PR DESCRIPTION
**Pull request by samwaseda**
I largely simplified the step to create directories and copy files. Hopefully the functionality is the same.

### CODE REVIEW

**Comment by samwaseda** [comment]
Sorry tests are failing so I take back the review requests

**Comment by samwaseda** [comment]
Ok it looks like it was just a stupid mistake. I guess now it's ready

File: pyiron_base/project/archiving/export_archive.py [hunk 1]
```diff
@@ -78,36 +59,21 @@ def copy_files_to_archive(
...
-                os.path.relpath(f, directory_to_transfer),
-            ),
+        copytree(
+            directory_to_transfer, dst, ignore=ignore_non_h5_files, dirs_exist_ok=True
         )
     if compressed:
         compress_dir(archive_directory)
 
 
+def ignore_non_h5_files(dir, files):
```
Comment (pmrv) [inline pyiron_base/project/archiving/export_archive.py:73]: Could be replaced with `shutil.ignore_patterns('*.h5')`, but aesthetic only.

Does this work with nested projects/child jobs? If I understand the docs correctly `copytree` passes the files and directories and if you "ignore" directories does not recurse into them, but I may just misunderstand.

**Review by pmrv — APPROVED** [review]
Left a nit, but mostly lgtm.

File: pyiron_base/project/archiving/export_archive.py (hunk 1 above)
Comment (samwaseda) [reply pyiron_base/project/archiving/export_archive.py:73]: > Could be replaced with shutil.ignore_patterns('*.h5'), but aesthetic only.

It's actually other way around: Only files ending with `.h5` are not to be ignored.

Comment (samwaseda) [reply pyiron_base/project/archiving/export_archive.py:73]: > Does this work with nested projects/child jobs? If I understand the docs correctly `copytree` passes the files and directories and if you "ignore" directories does not recurse into them, but I may just misunderstand.

I got a reply from ChatGPT:

### Example Directory Structure

Suppose you have the following directory structure:

```
source_directory/
|-- file1.txt
|-- file2.h5
|-- file3.py
|-- subdir/
    |-- file4.h5
    |-- file5.txt
```

After running the script, `destination_directory` will have the following structure:

```
destination_directory/
|-- file2.h5
|-- subdir/
    |-- file4.h5
```

Comment (pmrv) [reply pyiron_base/project/archiving/export_archive.py:73]: > > Could be replaced with shutil.ignore_patterns('*.h5'), but aesthetic only.
> 
> It's actually other way around: Only files ending with `.h5` are not to be ignored.

Ah, I read too quickly.

**Pull request merged**